pallets/ah-migrator/src/treasury.rs

```diff
@@ -0,0 +1,186 @@
+			count_good,
+			count_bad,
+		});
+		log::info!(target: LOG_TARGET, "Processed {} treasury messages", count_good);
+
+		Ok(())
+	}
+
+	fn do_process_treasury_message(message: RcTreasuryMessageOf<T>) -> Result<(), Error<T>> {
+		log::debug!(target: LOG_TARGET, "Processing treasury message: {:?}", message);
+
+		match message {
+			RcTreasuryMessage::ProposalCount(proposal_count) => {
+				pallet_treasury::ProposalCount::<T>::put(proposal_count);
+			},
+			RcTreasuryMessage::Proposals((proposal_index, proposal)) => {
+				pallet_treasury::Proposals::<T>::insert(proposal_index, proposal);
+			},
+			RcTreasuryMessage::Deactivated(deactivated) => {
+				pallet_treasury::Deactivated::<T>::put(deactivated);
+			},
+			RcTreasuryMessage::Approvals(approvals) => {
+				let approvals = BoundedVec::<_, <T as pallet_treasury::Config>::MaxApprovals>::defensive_truncate_from(approvals);
+				pallet_treasury::Approvals::<T>::put(approvals);
+			},
+			RcTreasuryMessage::SpendCount(spend_count) => {
+				treasury_alias::SpendCount::<T>::put(spend_count);
+			},
+			RcTreasuryMessage::Spends { id: spend_index, status: spend } => {
+				let treasury_alias::SpendStatus {
+					asset_kind,
+					amount,
+					beneficiary,
+					valid_from,
+					expire_at,
+					status,
+				} = spend;
+				let (asset_kind, beneficiary) =
+					T::RcToAhTreasurySpend::convert((asset_kind, beneficiary)).map_err(|_| {
+						defensive!(
+							"Failed to convert RC treasury spend to AH treasury spend: {:?}",
+							spend_index
+						);
+						Error::FailedToConvertType
+					})?;
+				let spend = treasury_alias::SpendStatus {
+					asset_kind,
+					amount,
+					beneficiary,
+					valid_from,
+					expire_at,
+					status,
+				};
+				log::debug!(target: LOG_TARGET, "Mapped treasury spend: {:?}", spend);
+				treasury_alias::Spends::<T>::insert(spend_index, spend);
+			},
+			// TODO: migrate with new sdk version
+			// RcTreasuryMessage::LastSpendPeriod(last_spend_period) => {
+			// 	pallet_treasury::LastSpendPeriod::<T>::put(last_spend_period);
+			// },
+			RcTreasuryMessage::Funds => {
+				Self::migrate_treasury_funds();
+			},
+		}
+
+		Ok(())
+	}
+
+	/// Migrate treasury funds.
+	///
+	/// Transfer all assets from old treasury account id on Asset Hub (account id derived from the
+	/// treasury pallet location on RC from the perspective of AH) to new account id on Asset Hub
+	/// (the treasury account id used on RC).
+	fn migrate_treasury_funds() {
+		let (old_account_id, assets) = T::TreasuryAccounts::get();
+		let account_id = pallet_treasury::Pallet::<T>::account_id();
+
+		// transfer all assets from old treasury account id to new account id
+		for asset in assets {
+			let reducible = T::Assets::reducible_balance(
+				asset.clone(),
+				&old_account_id,
+				Preservation::Expendable,
+				Fortitude::Polite,
+			);
+
+			match T::Assets::transfer(
+				asset.clone(),
+				&old_account_id,
+				&account_id,
+				reducible,
+				Preservation::Expendable,
+			) {
```

Comment:
```suggestion
			).defensive() {
```
Maybe

Reply:
I think I will leave just error, I do not wanna make this error break the builds. We will inspect all errors anyway.